src/pkg/auth/client.go

```diff
@@ -0,0 +1,249 @@
+package auth
+
+import (
+	"encoding/json"
+	"errors"
+	"net/http"
+	"net/url"
+	"time"
+
+	"github.com/golang-jwt/jwt/v5"
+	"github.com/google/uuid"
+)
+
+type ResponseType string
+
+const (
+	CodeResponseType  ResponseType = "code"
+	TokenResponseType ResponseType = "token"
+)
+
+type AuthorizeOptions struct {
+	pkce     bool
+	provider string
+	// scopes []string
+}
+
+type AuthorizeOption = func(*AuthorizeOptions)
+
+func WithPkce() AuthorizeOption {
+	return func(o *AuthorizeOptions) {
+		o.pkce = true
+	}
+}
+
+func WithProvider(provider string) AuthorizeOption {
+	return func(o *AuthorizeOptions) {
+		o.provider = provider
+	}
+}
+
+type AuthorizeResult struct {
+	state    string
+	verifier string
+	url      url.URL
+}
+
+type ExchangeSuccess struct {
+	Tokens
+}
+
+type RefreshOptions struct {
+	access string
+}
+
+type RefreshOption func(*RefreshOptions)
+
+func WithAccessToken(access string) RefreshOption {
+	return func(o *RefreshOptions) {
+		o.access = access
+	}
+}
+
+type RefreshSuccess struct {
+	*Tokens
+}
+
+type VerifyOptions struct {
+	refresh string
+}
+
+type VerifyOption func(*VerifyOptions)
+
+func WithRefreshToken(refresh string) VerifyOption {
+	return func(vo *VerifyOptions) {
+		vo.refresh = refresh
+	}
+}
+
+type Tokens struct {
+	AccessToken  string `json:"access_token"`
+	RefreshToken string `json:"refresh_token"`
+	// ExpiresIn   int    `json:"expires_in"`
```

Comment:
I think I remember reading that `ExpiresIn` would never be supported. If that's true, can we remove this line? If I'm wrong, maybe we can add a comment to describe our intentions?

Reply:
It *should* be supported, but only for the response (not the request). OpenAuth has a PR open for it.

Reply:
Can we link to that PR?

Reply:
https://github.com/toolbeam/openauth/pull/187